pkg/adapter/adapter.go

```diff
@@ -128,6 +126,21 @@ func (l listener) handleEvent() http.HandlerFunc {
 		var logger *zap.SugaredLogger
 
 		l.event = info.NewEvent()
+
+		// if repository auto configuration is enabled then check if its a valid event
+		if l.run.Info.Pac.AutoConfigureNewRepo {
+			detected, configuring, err := github.ConfigureRepository(ctx, l.run, request, string(payload), l.logger)
+			if detected {
+				if configuring && err != nil {
+					l.writeResponse(response, http.StatusAccepted, "configured")
+					return
+				}
+				l.logger.Infof("repository configuration detected: configuring: %v, err: %v", configuring, err)
+				l.writeResponse(response, http.StatusOK, "rejected")
```

Comment:
I am not understanding this code, is the info string wrong?

Reply:
if its a repository event -> `detected` is returned as `true`
then if the action is `created` -> `configuring` is returned as `true`

here if action is other than created then we return the response as ok but saying rejected.

Reply:
ah I think I messed it up here.. the status code too
its not a async execution..
* will return statusCreated is we configure it
* or just okay if we don't do anything

wdyt?

Reply:
status created sounds fine, did you look in the github documentation if they expect something back?

Reply:
on github we do see a response code, if pac is down, this will show unreachable here
![image](https://user-images.githubusercontent.com/55777192/196900795-c7b0805d-08f9-4183-a92c-b2b1b8bea445.png)